sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -39,19 +39,31 @@ public static HttpPipeline Build(ClientOptions options, params HttpPipelinePolic
             HttpPipelinePolicy[] perRetryPolicies,
             ResponseClassifier? responseClassifier)
         {
-            return Build(options, perCallPolicies, perRetryPolicies, responseClassifier, null);
+            var result = BuildInternal(options, perCallPolicies, perRetryPolicies, null, responseClassifier);
+            return new HttpPipeline(result.Transport, result.PerCallIndex, result.PerRetryIndex, result.Policies.ToArray(), result.Classifier);
         }
 
         /// <summary>
-        /// Creates an instance of <see cref="HttpPipeline"/> populated with default policies, customer provided policies from <paramref name="options"/> and client provided per call policies.
+        /// Creates an instance of <see cref="DisposableHttpPipeline"/> populated with default policies, customer provided policies from <paramref name="options"/> and client provided per call policies.
         /// </summary>
         /// <param name="options">The customer provided client options object.</param>
         /// <param name="perCallPolicies">Client provided per-call policies.</param>
         /// <param name="perRetryPolicies">Client provided per-retry policies.</param>
-        /// <param name="responseClassifier">The client provided response classifier.</param>
         /// <param name="defaultTransportOptions">The customer provided transport options which will be applied to the default transport.</param>
-        /// <returns>A new instance of <see cref="HttpPipeline"/></returns>
-        public static HttpPipeline Build(ClientOptions options, HttpPipelinePolicy[] perCallPolicies, HttpPipelinePolicy[] perRetryPolicies, ResponseClassifier? responseClassifier, HttpPipelineTransportOptions? defaultTransportOptions)
+        /// <param name="responseClassifier">The client provided response classifier.</param>
+        /// <returns>A new instance of <see cref="DisposableHttpPipeline"/></returns>
+        public static DisposableHttpPipeline Build(ClientOptions options, HttpPipelinePolicy[] perCallPolicies, HttpPipelinePolicy[] perRetryPolicies, HttpPipelineTransportOptions defaultTransportOptions, ResponseClassifier? responseClassifier)
```

Comment:
> HttpPipelineTransportOptions defaultTransportOptions,

Is defaultTransportOptions the best name for this?  It sounds a little like we're passing a parameter to set a default, which is counterintuitive.

Reply:
The reason for the name is that the options are applied to the transport we create with our default options applied (as opposed to with no option set). Do you have any naming suggestions?

Reply:
I can make a suggestion once I understand it a little better... can you break it down for me a bit?

Build returns a pipeline.  The pipeline can have one transport?  More than one transport?  I'm a bit confused about what is the default transport vs. what is the transport with no options.  Help?

Reply:
Sure thing - It's probably more accurate to make the distinction between the default transport and a custom transport. When no custom options are applied to a Transport in ClientOptions, a transport is created with our defaults set. Ex:

https://github.com/Azure/azure-sdk-for-net/blob/ac9c1d3f6a27536428070e7e38be8060a8535d06/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs#L39

https://github.com/Azure/azure-sdk-for-net/blob/ac9c1d3f6a27536428070e7e38be8060a8535d06/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs#L145-L156

This default instance is static, is considered the default transport, and is shared by all other clients that that did not configure a custom transport. If any custom Transport options are set, we do not apply our defaults and construct the custom transport with the supplied options.

The `defaultTransportOptions` are options applied to the default transport, rather than a totally custom transport.

Reply:
What happens if I pass defaultTransportOptions, but I've also specified a custom transport via client options?

Reply:
The options are not applied (and we log accordingly)

Reply:
Discussed offline and made some changes to the doc comments and name of the parameter.